community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -172,11 +171,10 @@ void assertOpen()
             throw new NotInTransactionException( "The statement has been closed." );
         }
 
-        Status terminationReason = transaction.getReasonIfTerminated();
-        if ( terminationReason != null )
+        transaction.getReasonIfTerminated().ifPresent( reason ->
```

Comment:
`assertOpen` is called very very often. Isn't it a bad idea to create this lambda for every call to it? Just because Option do exist now should it be used in all places where null was used IMHO. Even when speculating about escape analysis and how the JVM may avoid creating that instance it feels unnecessary to even give it a chance to create it and potentially contribute to garbage creep.